### PR TITLE
fix(metrics): Do not crash when limiting all strings in a message [INGEST-1380]

### PR DESCRIPTION
**Pull request by untitaker**
If the postgres indexer backend determines that there are no strings to
be written to postgres because of rate limits, it returns early, just as
we do when there are no strings to be read from postgres. In that
early-return path, we were accidentally omitting the KeyResults for the
dropped strings + metadata that indicates that they were, causing a
KeyError in StringBatch when attempting to access the result.

We didn't even drop data this way, we just crashed processing a message
that we would've dropped anyway.

Also add a few more tags that are useful to determine impact of errors:
A usecase tag to see if an indexer crash affects releasehealth or
performance, and a tag for the organization id to quickly determine
which organization is the most heavily rate limited.

### CODE REVIEW

File: src/sentry/sentry_metrics/indexer/postgres_v2.py [hunk 1]
```diff
@@ -144,11 +144,19 @@ def bulk_record(
...
+                    fetch_type=dropped_string.fetch_type,
+                    fetch_type_ext=dropped_string.fetch_type_ext,
+                )
+
             if filtered_db_write_keys.size == 0:
                 indexer_cache.set_many(new_results_to_cache, use_case_id.value)
-                return cache_key_results.merge(db_read_key_results)
-
-            rate_limited_write_results = writes_limiter_state.dropped_strings
+                return cache_key_results.merge(db_read_key_results).merge(db_write_key_results)
```
Comment (MeredithAnya) [inline src/sentry/sentry_metrics/indexer/postgres_v2.py:159]: so does this mean `db_write_key_results` can be a combination of dropped write keys and then also successful results? Would it be weird to have another `KeyResults` that is `db_dropped_write_keys`? Maybe that's too many merges at the bottom 

```python
return cache_key_results.merge(db_read_key_results).merge(db_write_key_results).merge(db_dropped_key_results)
```

Comment (untitaker) [reply src/sentry/sentry_metrics/indexer/postgres_v2.py]: that makes sense, fixed. curious how this will affect performance...

Comment (MeredithAnya) [reply src/sentry/sentry_metrics/indexer/postgres_v2.py]: yeah wasn't sure about the overhead of creating a new `KeyResults` would be... maybe we will find out

**Review by MeredithAnya — APPROVED** [review]
LGTM

**Pull request merged**